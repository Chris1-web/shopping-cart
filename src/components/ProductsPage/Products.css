/* cart relative to main section */
#root {
  position: relative;
}

.cart {
  background-color: white;
  position: fixed;
  right: 0;
  width: 7rem;
  height: 7rem;
  border: 0.2rem solid black;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 1rem 0;
}

.cart img {
  width: 5rem;
  height: 5rem;
}

.cart p {
  position: absolute;
  top: 50%;
  right: 50%;
  width: 3rem;
  height: 3rem;
  background-color: pink;
  border-radius: 50%;
  text-align: center;
  font-size: 2rem;
}

.products {
  padding: 2rem 1.5rem;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(28rem, 1fr));
  gap: 2rem;
  background-color: #f5f5f5;
}

.products img {
  width: 30rem;
  height: 18rem;
}

.product-card {
  height: 32rem;
  width: 30rem;
  background-color: white;
  display: flex;
  flex-direction: column;
  align-self: center;
  justify-self: center;
  transition: all 0.5s;
}

.product-card h3,
.product-card p {
  padding: 0 1rem;
}

.product-card h3 {
  font-size: 2rem;
}

.product-card p {
  font-size: 2rem;
}

.product-card .add-to-cart-btn {
  margin: 1.5rem auto;
  width: 95%;
  padding: 1rem;
  font-size: 1.6rem;
  border: 0.2rem solid #0c4b23;
  color: #0c4b23;
  background-color: white;
  transition: all 0.4s;
}

.product-card .add-to-cart-btn:hover {
  color: white;
  background-color: #0c4b23;
  box-shadow: 0.5rem 0.2rem 0.5rem #0c4b23;
}

.product-card:hover {
  box-shadow: 0.6rem 0.3rem 0.6rem #0c4b23;
  cursor: pointer;
}

input[type="number"]::-webkit-inner-spin-button,
input[type="number"]::-webkit-outer-spin-button {
  -webkit-appearance: none;
}

.product-card input[type="number"] {
  width: 20%;
  height: 2rem;
  border: none;
  border-radius: 2px;
  text-align: center;
  box-shadow: 0.2rem 0.2rem 0.5rem white;
}

.product-card .form {
  display: flex;
  align-items: center;
  height: 2rem;
  padding: 0 1rem;
}

.product-card .form button {
  padding: 0 1rem;
  height: 2rem;
  background-color: #3d6f4f;
  color: white;
  border: none;
  border-radius: 2px;
  box-shadow: 0.2rem 0.2rem 0.5rem #0c4b23;
  font-size: 2rem;
}

.product-card .form button:hover {
  color: white;
  background-color: #0c4b23;
  box-shadow: 0.5rem 0.2rem 0.5rem #0c4b23;
}

.add-to-cart-btn {
  width: 100%;
}

.product-card button:disabled {
  background-color: #6d937b;
  color: #cccccc;
}

.product-card button:disabled:hover {
  background-color: #6d937b;
  color: #cccccc;
  cursor: not-allowed;
  box-shadow: none;
}
