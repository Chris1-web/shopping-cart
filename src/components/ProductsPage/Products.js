import { useEffect, useState } from "react";
import Cart from "../cartPage/Cart";
import "./Products.css";
import ProductCard from "./ProductCard";
import cheapRoadBicycle from "../images/cheap-Road-Bike.jpg";
import KidsBicycle from "../images/Kids-Bicycle.jpg";
import MarketBicycle from "../images/Market-Bicycle.jpg";
import RenaissanceBicycle from "../images/Renaissance-Street-Bicycle.jpg";
import SafeRoadBicycle from "../images/safe-road-bicycle.jpg";
import TouristBicycle from "../images/Tourist-Bicycle.jpg";
import TwinBicycle from "../images/Twin-Bicycle.jpg";
import MountainBicycle from "../images/mountainBike.jpg";
import AdultChoiceBicycle from "../images/AdultChoice.jpg";
import cartLogo from "../images/cart-logo.png";

const productsImages = [
  {
    name: "Mountain Bicycle",
    src: MountainBicycle,
    price: 1000,
    quantity: 1,
  },
  {
    name: "Safe Road Bicycle",
    src: SafeRoadBicycle,
    price: 200,
    quantity: 1,
  },
  {
    name: "Renaissance Bicycle",
    src: RenaissanceBicycle,
    price: 1000,
    quantity: 1,
  },
  {
    name: "Kid's Bicycle",
    src: KidsBicycle,
    price: 200,
    quantity: 1,
  },
  {
    name: "Adult Choice Bicycle",
    src: AdultChoiceBicycle,
    price: 500,
    quantity: 1,
  },
  {
    name: "Twin Bicycle",
    src: TwinBicycle,
    price: 5000,
    quantity: 1,
  },
  {
    name: "Tourist Bicycle",
    src: TouristBicycle,
    price: 300,
    quantity: 1,
  },
  {
    name: "Cheap Road Bicycle",
    src: cheapRoadBicycle,
    price: 150,
    quantity: 1,
  },
  {
    name: "Market Bicycle",
    src: MarketBicycle,
    price: 500,
    quantity: 1,
  },
];

export default function Products() {
  const [products, setProducts] = useState([]);
  const [showCart, setShowCart] = useState(false);
  const [cartProduct, setCartProduct] = useState([]);

  useEffect(() => {
    setProducts([...productsImages]);
  }, []);

  const addProductToCart = (product) => {
    console.log(product);
    console.log(productsImages);
  };

  const changeQuantity = (clickedProduct, newQuantity) => {
    if (!newQuantity || newQuantity === 0) return;
    const chosenProduct = products.map((product) => {
      if (product.name === clickedProduct.name) {
        product.quantity = +newQuantity;
        return product;
      }
      return product;
    });
    console.log(clickedProduct);
    console.log(chosenProduct);
  };

  const toggleCart = () => {
    showCart ? setShowCart(false) : setShowCart(true);
  };

  return (
    <>
      <div className="cart" onClick={toggleCart}>
        <img src={cartLogo} alt="page logo" className="cart-logo" />
        <p className="cart-product-number">1</p>
      </div>
      <main className="products">
        {products.map((product, index) => (
          <ProductCard
            product={product}
            changeQuantity={changeQuantity}
            addProductToCart={addProductToCart}
            key={index}
          />
        ))}
        {showCart && <Cart toggleCart={toggleCart} />}
      </main>
    </>
  );
}
