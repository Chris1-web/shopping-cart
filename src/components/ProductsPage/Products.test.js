import { render } from "@testing-library/react";
import Products from "../ProductsPage/Products";

describe("App component", () => {
  it("renders products page", () => {
    // since screen does not have the container property, we'll destructure render to obtain container for this test
    const { container } = render(<Products />);
    expect(container).toMatchSnapshot();
  });
});
