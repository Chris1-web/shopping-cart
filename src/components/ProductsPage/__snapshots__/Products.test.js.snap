// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders products page 1`] = `
<div>
  <div
    class="cart"
  >
    <img
      alt="page logo"
      class="cart-logo"
      src="cart-logo.png"
    />
    <p
      class="cart-product-number"
    >
      0
    </p>
  </div>
  <main
    class="products"
  >
    <article
      class="product-card"
    >
      <img
        alt="Mountain Bicycle"
        class="product-card-image"
        src="mountainBike.jpg"
      />
      <h3>
        Mountain Bicycle
      </h3>
      <p>
        $
        1000
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Safe Road Bicycle"
        class="product-card-image"
        src="safe-road-bicycle.jpg"
      />
      <h3>
        Safe Road Bicycle
      </h3>
      <p>
        $
        200
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Renaissance Bicycle"
        class="product-card-image"
        src="Renaissance-Street-Bicycle.jpg"
      />
      <h3>
        Renaissance Bicycle
      </h3>
      <p>
        $
        1000
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Kid's Bicycle"
        class="product-card-image"
        src="Kids-Bicycle.jpg"
      />
      <h3>
        Kid's Bicycle
      </h3>
      <p>
        $
        200
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Adult Choice Bicycle"
        class="product-card-image"
        src="AdultChoice.jpg"
      />
      <h3>
        Adult Choice Bicycle
      </h3>
      <p>
        $
        500
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Twin Bicycle"
        class="product-card-image"
        src="Twin-Bicycle.jpg"
      />
      <h3>
        Twin Bicycle
      </h3>
      <p>
        $
        5000
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Tourist Bicycle"
        class="product-card-image"
        src="Tourist-Bicycle.jpg"
      />
      <h3>
        Tourist Bicycle
      </h3>
      <p>
        $
        300
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Cheap Road Bicycle"
        class="product-card-image"
        src="cheap-Road-Bike.jpg"
      />
      <h3>
        Cheap Road Bicycle
      </h3>
      <p>
        $
        150
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
    <article
      class="product-card"
    >
      <img
        alt="Market Bicycle"
        class="product-card-image"
        src="Market-Bicycle.jpg"
      />
      <h3>
        Market Bicycle
      </h3>
      <p>
        $
        500
      </p>
      <button
        class="add-to-cart-btn"
      >
        Add To Cart
      </button>
    </article>
  </main>
</div>
`;
